usr/share/rear/layout/prepare/GNU/Linux/130_include_mount_filesystem_code.sh

```diff
@@ -69,10 +70,53 @@ mount_fs() {
             # that different subvolume needs to be first created, then set to be the default subvolume, and
             # finally that btrfs filesystem needs to be unmounted and mounted again so that in the end
             # that special different default subvolume is mounted at the mountpoint $TARGET_FS_ROOT$mountpoint.
-            # All btrfs subvolume handling happens in the btrfs_subvolumes_setup function in 130_include_mount_subvolumes_code.sh
-            # For a plain btrfs filesystem without subvolumes the btrfs_subvolumes_setup function does nothing.
-            # Call the btrfs_subvolumes_setup function for the btrfs filesystem that was mounted above:
-            btrfs_subvolumes_setup $device $mountpoint $mountopts
+            # All btrfs subvolume handling happens in the btrfs_subvolumes_setup_SLES function in 130_include_mount_subvolumes_code.sh
+            # or in the btrfs_subvolumes_setup_generic function in 131_include_mount_subvolumes_code.sh.
+            # For a plain btrfs filesystem without subvolumes the btrfs_subvolumes_setup_* functions do nothing.
+            # Call the right btrfs_subvolumes_setup_* function for the btrfs filesystem that was mounted above via an
+            # artificial 'for' clause that is run only once to be able to 'continue' with the code after it:
+            for dummy in "once" ; do
+                # When both BTRFS_SUBVOLUME_GENERIC_SETUP and BTRFS_SUBVOLUME_SLES_SETUP are explicity set to false
+                # no special btrfs subvolumes setup is done which may lead to a falsely recreated system
+                # but we do what the user has explicity specified:
+                if is_false "$BTRFS_SUBVOLUME_GENERIC_SETUP" && is_false "$BTRFS_SUBVOLUME_SLES_SETUP" ; then
+                    LogPrint "Skipping btrfs subvolumes setup for $device on $mountpoint (BTRFS_SUBVOLUME_GENERIC_SETUP and BTRFS_SUBVOLUME_SLES_SETUP false)"
+                    continue
+                fi
+                if is_true "$BTRFS_SUBVOLUME_GENERIC_SETUP" ; then
+                    LogPrint "Doing generic btrfs subvolumes setup for $device on $mountpoint (BTRFS_SUBVOLUME_GENERIC_SETUP true)"
+                    btrfs_subvolumes_setup_generic $device $mountpoint $mountopts
```

Comment:
$mountopts is not used by the callee and can safely be dropped here (likewise in subsequent lines).
```suggestion
                    btrfs_subvolumes_setup_generic $device $mountpoint
```

Reply:
Done via
https://github.com/rear/rear/pull/2080/commits/cd2bd2e60a6824b2244ef91abb34f8d239dcb500